### ISSUE TITLE
Bizarre error about unable to load "package.json" when building with "remix build"

### ISSUE THREAD

**Issue by ymc9** [remix-run/remix#5755]
### What version of Remix are you using?

1.14.1

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

1. Clone project from "https://github.com/ymc9/remix-build-issue"
2. npm run build

The project was created with indie-stack and with just one additional dependency installed: "@zenstackhq/runtime".

If in the route, I call "getVersion" function from the package, build fails with bizarre behavior. Please check `loader` in `/app/routes/notes.tsx` file.

However, the "getVersion" function implementation is super straightforward:

```js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getVersion = void 0;
/* eslint-disable @typescript-eslint/no-var-requires */
function getVersion() {
    try {
        return require('./package.json').version;
    }
    catch (_a) {
        return require('../package.json').version;
    }
}
exports.getVersion = getVersion;
//# sourceMappingURL=version.js.map
```

I'm not sure what magic remix does during building. There's no reason why the first line of `require('./package.json')` should fail IMO.

### Expected Behavior

Build succeeds.

### Actual Behavior

Build fails with the following error:

```
/Users/yiming/git/playground/remix-build-issue/node_modules/esbuild/lib/main.js:1591
  let error = new Error(`${text}${summary}`);
              ^

Error: Build failed with 1 error:
node_modules/@zenstackhq/runtime/version.js:10:23: ERROR: Could not resolve "../package.json"
    at failureErrorWithLog (/Users/yiming/git/playground/remix-build-issue/node_modules/esbuild/lib/main.js:1591:15)
    at /Users/yiming/git/playground/remix-build-issue/node_modules/esbuild/lib/main.js:1047:28
    at runOnEndCallbacks (/Users/yiming/git/playground/remix-build-issue/node_modules/esbuild/lib/main.js:1463:61)
    at buildResponseToResult (/Users/yiming/git/playground/remix-build-issue/node_modules/esbuild/lib/main.js:1045:7)
    at /Users/yiming/git/playground/remix-build-issue/node_modules/esbuild/lib/main.js:1157:14
    at responseCallbacks.<computed> (/Users/yiming/git/playground/remix-build-issue/node_modules/esbuild/lib/main.js:694:9)
    at handleIncomingPacket (/Users/yiming/git/playground/remix-build-issue/node_modules/esbuild/lib/main.js:749:9)
    at Socket.readFromStdout (/Users/yiming/git/playground/remix-build-issue/node_modules/esbuild/lib/main.js:670:7)
    at Socket.emit (node:events:527:28)
    at addChunk (node:internal/streams/readable:315:12)
    at readableAddChunk (node:internal/streams/readable:289:9)
    at Socket.Readable.push (node:internal/streams/readable:228:10)
    at Pipe.onStreamRead (node:internal/stream_base_commons:190:23) {
  errors: [
    {
      detail: undefined,
      id: '',
      location: {
        column: 23,
        file: 'node_modules/@zenstackhq/runtime/version.js',
        length: 17,
        line: 10,
        lineText: "        return require('../package.json').version;",
        namespace: '',
        suggestion: ''
      },
      notes: [],
      pluginName: '',
      text: 'Could not resolve "../package.json"'
    }
  ],
  warnings: []
}
ERROR: "build:remix" exited with 1.
```

**Comment by ymc9** [comment]
What's strange is that, if I just move the call to "getVersion" into a helper function exported from `/app/models/note.server.ts`, the build error is gone ... if feels that remix build is doing something special for code directly under "routes" folder?

**Comment by akamfoad** [comment]
Please checkout the [**Server code in client bundles**](https://remix.run/docs/en/main/pages/gotchas#md-server-code-in-client-bundles) in Remix docs, you'll find out why this is happening.

TL;DR when you import a server-only intended code directly in a route file, it might get included inside the browser bundle where that import is not available and inevitably your application crashes, Remix catches this first hand, when you import it from a `*.server.*` file, the bundlers that Remix use know this code is intended to be on server and run there only so they package it accordingly.

When remix builds your app, in development and in production, it creates to types of outputs one is for the server, and one for the browser, your server code never makes it into the browser and you send minimal code that is necessary for your app to work there. In a typical Remix app:
- Server Bundle located at : `<ROOT_DIR>/build/`
- Browser Bundle locate at: `<ROOT_DIR>/public/build/`

**Comment by ymc9** [comment]
Thanks for the explanation. I thought this was related to bundling server code into client side but just surprised that it triggered a runtime error at build time.

**Comment by brophdawg11** [comment]
This sounds like it's expected and has an easy-enough work around so I'm going to close this out
